packages/nbdime/src/common/mergeview.ts

```diff
@@ -143,563 +341,404 @@ function createNbdimeMergeView(
-            // as there is no way to re-pick them
-            s.decision.customDiff = [];
-          }
-        }
-        if (ss.length === 0) {
-          // All decisions empty, remove picker
-          // In these cases, there should only be one picker, on base
-          // so simply remove the one we have here
-          instance.setGutterMarker(line, GUTTER_PICKER_CLASS, null);
-        }
-      } else if (gutter === GUTTER_CONFLICT_CLASS) {
-        for (let s of ss) {
-          s.decision.conflict = false;
-        }
-      }
-      for (let dv of this.baseEditor.state.diffViews as DiffView[]) {
-        if (dv.model instanceof DecisionStringDiffModel) {
-          dv.model.invalidate();
-        }
-        dv.forceUpdate('full');
-      }
-    }
-  }
-
-  protected registerScroll(): void {
-    let self = this;
-    this.baseEditor.on('scroll', function() {
-      self.syncScroll(EventDirection.OUTGOING);
-    });
-    this.ownEditor.on('scroll', function() {
-      self.syncScroll(EventDirection.INCOMING);
-    });
-  }
-
-  /**
+ /**
    * Sync scrolling between base and own editors. `type` is used to indicate
    * which editor is the source, and which editor is the destination of the sync.
    */
-  protected syncScroll(type: EventDirection): void {
+  protected syncScroll (srcEditor: EditorView, destEditor: EditorView): void {
     if (this.modelInvalid()) {
       return;
     }
-    if (!this.lockScroll) {
-      return;
-    }
-    // editor: What triggered event, other: What needs to be synced
-    let editor: CodeMirror.Editor;
-    let other: CodeMirror.Editor;
-    if (type === EventDirection.OUTGOING) {
-      editor = this.baseEditor;
-      other = this.ownEditor;
-    } else {
-      editor = this.ownEditor;
-      other = this.baseEditor;
-    }
-
-    if (editor.state.scrollSetBy === this) {
-      editor.state.scrollSetBy = null;
-      return;
-    }
-
-    // Position to update to
-    other.state.scrollPosition = editor.getScrollInfo();
-
-    // If ticking, we already have a scroll queued
-    if (other.state.scrollTicking) {
-      return;
-    }
-
-    let sInfo = other.getScrollInfo();
-    // Don't queue an event if already synced.
-    if (other.state.scrollPosition.top === sInfo.top &&
-        other.state.scrollPosition.left === sInfo.left) {
-      return;
-    }
-    // Throttle by requestAnimationFrame().
-    // If event is outgoing, this will lead to a one frame delay of other DiffViews
-    let self = this;
-    window.requestAnimationFrame(function() {
-      other.scrollTo(other.state.scrollPosition.left, other.state.scrollPosition.top);
-      other.state.scrollTicking = false;
-      other.state.scrollSetBy = self;
+    let srcScroller = srcEditor.scrollDOM;
+    let destScroller = destEditor.scrollDOM;
+    srcScroller.addEventListener("scroll", (event) => {
+      window.requestAnimationFrame(function() {destScroller.scrollLeft = srcScroller.scrollLeft;})
     });
```

Comment:
It would be more natural to have this function add the eventListener to both editors instead of having to call it twice.

Reply:
Thanks. This has been taken into account.